tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Kafka/KafkaHelper.cs

```diff
@@ -94,6 +95,54 @@ internal static class KafkaHelper
             return scope;
         }
 
+        private static long TryGetSize(object obj)
+        {
+            if (obj == null)
+            {
+                return 0;
+            }
+
+            if (obj is byte[] bytes)
+            {
+                return bytes.Length;
+            }
+
+            if (obj is string str)
+            {
+                return Encoding.UTF8.GetByteCount(str);
+            }
+
+            return 0;
+        }
+
+        private static long GetMessageSize(IMessage message)
+        {
+            long size = 0;
+            if (message == null)
```

Comment:
`message` will never be null, due to the way duck typing works. instead, you need to do something like this:

```suggestion
            if (((IDuckType)message).Instance is null)
```